browser/ui/views/frame/split_view/brave_multi_contents_view_mini_toolbar.cc

```diff
@@ -0,0 +1,119 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/ui/views/frame/split_view/brave_multi_contents_view_mini_toolbar.h"
+
+#include "base/i18n/rtl.h"
+#include "brave/browser/ui/color/brave_color_id.h"
+#include "brave/browser/ui/views/frame/brave_contents_view_util.h"
+#include "brave/browser/ui/views/frame/split_view/brave_contents_container_view.h"
+#include "brave/components/vector_icons/vector_icons.h"
+#include "chrome/browser/ui/views/frame/top_container_background.h"
+#include "third_party/skia/include/core/SkMatrix.h"
+#include "third_party/skia/include/core/SkPath.h"
+#include "ui/base/metadata/metadata_impl_macros.h"
+#include "ui/color/color_provider.h"
+#include "ui/gfx/canvas.h"
+#include "ui/gfx/vector_icon_types.h"
+#include "ui/views/layout/flex_layout.h"
+
+namespace {
+constexpr int kMiniToolbarContentPadding = 4;
+constexpr int kMiniToolbarOutlineCornerRadius = 8;
+}  // namespace
+
+const gfx::VectorIcon& GetMoreVerticalIcon() {
+  return kLeoMoreVerticalIcon;
+}
+
+BraveMultiContentsViewMiniToolbar::~BraveMultiContentsViewMiniToolbar() =
+    default;
+
+void BraveMultiContentsViewMiniToolbar::UpdateState(bool is_active) {
+  MultiContentsViewMiniToolbar::UpdateState(is_active);
+
+  if (!GetVisible()) {
+    return;
+  }
+
+  is_active_ = is_active;
+  stroke_color_ = is_active_ ? kColorBraveSplitViewActiveWebViewBorder
+                             : kColorBraveSplitViewInactiveWebViewBorder;
+
+  const gfx::Insets active_interior_margins = gfx::Insets::TLBR(
+      kMiniToolbarOutlineCornerRadius + kMiniToolbarContentPadding,
+      kMiniToolbarOutlineCornerRadius + kMiniToolbarContentPadding,
+      kMiniToolbarContentPadding, GetOutlineThickness() * 2);
+
+  const gfx::Insets inactive_interior_margins = gfx::Insets::TLBR(
+      kMiniToolbarOutlineCornerRadius + kMiniToolbarContentPadding,
+      kMiniToolbarOutlineCornerRadius * 2, kMiniToolbarContentPadding,
+      GetOutlineThickness());
+
+  static_cast<views::FlexLayout*>(GetLayoutManager())
+      ->SetInteriorMargin(is_active ? active_interior_margins
+                                    : inactive_interior_margins);
+}
+
+void BraveMultiContentsViewMiniToolbar::OnPaint(gfx::Canvas* canvas) {
+  // Paint the mini toolbar background to match the toolbar.
```

Comment:
It'd be great if we say that MultiContentsViewMiniToolbar's implementation intentionally.

```suggestion
  // Bypassing MultiContentsViewMiniToolbar::OnPaint() and Paint the mini toolbar background to match the toolbar.
```